### ISSUE TITLE
initial={true} on tab page doesn't work.

### ISSUE THREAD

**Issue by chetstone** [aksonov/react-native-router-flux#62]
When I set initial={true} on Tab 3, the view initially is set to Tab 3, but the TabBar icon still shows Tab 1 is selected.

This is simple to reproduce in the current example. I have it in my repo [here](https://github.com/chetstone/react-native-router-flux/blob/13f5000319b0ece222f69e7b9ee07f792b21d367/Example/Example.js#L50)

**Comment by aksonov** [comment]
Please try 'selected' not initial. Initial is only for root initial route.

**Comment by chetstone** [comment]
Hmmm. That makes it worse. Now both Tab1 and Tab3 are highlighted in the tab bar, and the view is of Tab1.

**Comment by aksonov** [comment]
Then that is bug (selected), i will look at it, thanks.

**Comment by aksonov** [comment]
And you are right, probably i should do select it with 'initial', to avoid complexity.

**Comment by aksonov** [comment]
@chetstone could you look into this issue and fix it?

**Comment by chetstone** [comment]
I spent some time last night and tried a couple of things but the code was
very hard to follow. If you're thinking of refactoring, maybe this should
wait. Refactoring may be a good idea: my feeling (could be wrong) is that
processing the children of <Router> could be different. Somehow I think it
should not matter if the Router is a TabBar router or not --- any Router
(or nested router) should be treated the same, e.g. it could have 'initial'
defined.

What is different about a tab bar router is that it has to deal with an
extra component (TabBar), and the actions are switch instead of push/pop.
Otherwise the same.

**Comment by smallsaucepan** [comment]
@aksonov @chetstone Thought I might look at this. Do either of you have any changes pending?

**Comment by aksonov** [comment]
@smallsaucepan No, you could work on it.

**Comment by aksonov** [comment]
@smallsaucepan Any update for this?

**Comment by chetstone** [comment]
I finally had some time to look at this again. @aksonov, the good news is that it looks like the structure of your code has improved a lot with the refactor. The bad news is I haven't gotten any smarter, and I still can't figure it out. 

It looks like the error is caused by Tabbar:getChildrenState() being called with a null selectedRoute parameter on initial render. I've figured out a couple of ways to fix that --- the easiest for experimentation is just to add the prop `initial="tab4"` to tabbar router like this:
```
<Router footer={TabBar} showNavigationBar={false} initial="tab4" >
```

Then I get the error:
```
Cannot find router for route=tab4 current router=root
```

and I'm stuck trying to figure that out.

**Comment by aksonov** [comment]
Should be fixed with latest version, please check.

**Comment by chetstone** [comment]
Seems to be fixed in 2.3.6. Thank you!
